easybuild/framework/easyblock.py

```diff
@@ -1421,6 +1421,32 @@ def make_module_extra(self, altroot=None, altversion=None):
                                      value, type(value))
             lines.append(self.module_generator.append_paths(key, value, allow_abs=self.cfg['allow_append_abs_path']))
 
+        # Add automatic PYTHONPATH or EBPYTHONPREFIXES if they aren't already present and python paths exist
+        if self.name not in ['Python', 'Miniconda', 'Anaconda']:  # nothing extra needed for base python installations
+            # Exclude symlinks lib -> lib64 or vice verse to avoid duplicates
+            python_paths = []
+            if not os.path.islink(f'{self.installdir}/lib'):
+                python_paths += [path[len(self.installdir)+1:]
+                                 for path in glob.glob(f'{self.installdir}/lib/python*/site-packages')
+                                 if re.match(self.installdir + r'/lib/python\d+\.\d+/site-packages', path)]
+            if not os.path.islink(f'{self.installdir}/lib64'):
+                python_paths += [path[len(self.installdir)+1:]
+                                 for path in glob.glob(f'{self.installdir}/lib64/python*/site-packages')
+                                 if re.match(self.installdir + r'/lib64/python\d+\.\d+/site-packages', path)]
+
+            use_ebpythonprefixes = 'Python' in self.cfg.dependencies(runtime_only=True) and \
+                build_option('prefer_ebpythonprefixes') and self.cfg['prefer_ebpythonprefixes']
+
+            if len(python_paths) > 1 and not use_ebpythonprefixes:
+                raise EasyBuildError('Multiple python paths requires EBPYTHONPREFIXES: ' + ', '.join(python_paths))
+            elif python_paths:
```

Comment:
```suggestion
            if python_paths:
```

This is (as far as I have found, having tested just about every type of easyconfig i could find), which has an actual issue with the code right now

EasyBuild itself generates both:
```
/apps/Test2/software/EasyBuild/4.9.2/lib/python3.6/site-packages/
/apps/Test2/software/EasyBuild/4.9.2/lib64/python3.6/site-packages/
```
while the latter is empty, my glob'ing still picks it up, and having multiple PYTHONPATH's doesn't make any sense, this errors out. 
There is technically nothing wrong here, those paths are both compatible with python3.6, so they would be fine to use.

I can just remove that extra check. Simplifies the code and you simply get what you ask for. If you have multiple paths for some reason, they are both added, no problem. This is just an extra check we added that were never cared about before, and it's in no way a requirement for this PR.

On the other hand, it's literally *only* EasyBuild that does this, and I think it shouldn't duplicate the installation directories. Perhaps it should be creating a lib <-> lib64 symlink like just about everything else?

I could also make the code more robust, checking if it's really the same python version etc...

Reply:
I removed this check for now, to keep this as simple as possible. This check could be introduced later if we want it (but we never checked for this in the past, so this change really has nothing to do with EBPYTHONPREFIXES, especially considering it actually only is problem when EBPYTHONPREFIXES *isn't* used)